plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMaterializedViewDefinition.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
+import io.trino.spi.type.TypeId;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+/*
+ * Serializable version of ConnectorMaterializedViewDefinition stored by iceberg connector
+ */
+public class IcebergMaterializedViewDefinition
```

Comment:
keep the abstraction simple. If only iceberg is using this and it had broken serde, move it to `trino-iceberg`. If we think we might use it in Hive, move it to `trino-hive` and call it `TrinoMaterializedViewDefinition`.

But I don't think we will support Trino MVs in `trino-hive` soon since `storageTable` is kind of implementation specific